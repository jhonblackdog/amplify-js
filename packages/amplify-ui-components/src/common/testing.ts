export const pixelThreshold = 10;
