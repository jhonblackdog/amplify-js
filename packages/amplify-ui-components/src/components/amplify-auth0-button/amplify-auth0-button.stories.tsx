export default {
  title: 'amplify-auth0-button',
};

export const defaults = () => `<amplify-auth0-button />`;
