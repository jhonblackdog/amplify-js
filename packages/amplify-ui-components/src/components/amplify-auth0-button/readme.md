# amplify-auth0-button

<!-- Auto Generated Below -->

## Properties

| Property | Attribute | Description                                                           | Type                                                        | Default     |
| -------- | --------- | --------------------------------------------------------------------- | ----------------------------------------------------------- | ----------- |
| `config` | --        | See: https://auth0.com/docs/libraries/auth0js/v9#available-parameters | `{ [key: string]: any; clientID: string; domain: string; }` | `undefined` |

## Dependencies

### Used by

- [amplify-federated-buttons](../amplify-federated-buttons)

### Depends on

- [amplify-sign-in-button](../amplify-sign-in-button)

### Graph

```mermaid
graph TD;
  amplify-auth0-button --> amplify-sign-in-button
  amplify-federated-buttons --> amplify-auth0-button
  style amplify-auth0-button fill:#f9f,stroke:#333,stroke-width:4px
```

---

_Built with [StencilJS](https://stenciljs.com/)_
