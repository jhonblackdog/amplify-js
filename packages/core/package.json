{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
