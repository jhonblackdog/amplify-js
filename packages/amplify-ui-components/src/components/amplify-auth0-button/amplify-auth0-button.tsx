import { Component, h, Prop } from '@stencil/core';

import { SIGN_IN_WITH_AUTH0 } from '../../common/constants';
import { FederatedConfig } from '../../common/types/auth-types';

@Component({
  tag: 'amplify-auth0-button',
  shadow: false,
})
export class AmplifyAuth0Button {
  /** See: https://auth0.com/docs/libraries/auth0js/v9#available-parameters */
  @Prop() config: FederatedConfig['auth0Config'];

  signInWithAuth0(event) {
    event.preventDefault();

    throw new Error('Not implemented');
  }

  render() {
    return (
      <amplify-sign-in-button onClick={event => this.signInWithAuth0(event)} provider="auth0">
        <script src="https://cdn.auth0.com/js/auth0/9.11/auth0.min.js"></script>

        <svg slot="icon" id="artwork" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 193.7 216.6">
          <path
            d="M189,66.9L167.2,0H96.8l21.8,66.9H189z M96.8,0H26.5L4.8,66.9h70.4L96.8,0z M4.8,66.9L4.8,66.9	c-13,39.9,1.2,83.6,35.2,108.3l21.7-66.9L4.8,66.9z M189,66.9L189,66.9l-57,41.4l21.7,66.9l0,0C187.7,150.6,201.9,106.8,189,66.9	L189,66.9z M39.9,175.2L39.9,175.2l56.9,41.4l56.9-41.4l-56.9-41.4L39.9,175.2z"
            fill="#FFF"
          />
        </svg>

        {SIGN_IN_WITH_AUTH0}
      </amplify-sign-in-button>
    );
  }
}
