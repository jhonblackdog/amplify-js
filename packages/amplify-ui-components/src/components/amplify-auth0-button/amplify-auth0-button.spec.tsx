import { newSpecPage } from '@stencil/core/testing';

import { AmplifyAuth0Button } from './amplify-auth0-button';
import * as stories from './amplify-auth0-button.stories';

const {
  default: { title },
  ...specs
} = stories;

const components = [AmplifyAuth0Button];

describe(title, () => {
  describe('stories', () => {
    Object.entries(specs).forEach(([name, spec]) => {
      it(name, async () => {
        const page = await newSpecPage({ components, html: spec() });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
