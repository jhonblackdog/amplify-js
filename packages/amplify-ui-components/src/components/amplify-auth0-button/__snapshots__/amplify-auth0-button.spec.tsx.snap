// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-auth0-button stories defaults 1`] = `
<amplify-auth0-button>
  <amplify-sign-in-button provider="auth0">
    <script src="https://cdn.auth0.com/js/auth0/9.11/auth0.min.js"></script>
    <svg id="artwork" slot="icon" viewBox="0 0 193.7 216.6" xmlns="http://www.w3.org/2000/svg">
      <path d="M189,66.9L167.2,0H96.8l21.8,66.9H189z M96.8,0H26.5L4.8,66.9h70.4L96.8,0z M4.8,66.9L4.8,66.9	c-13,39.9,1.2,83.6,35.2,108.3l21.7-66.9L4.8,66.9z M189,66.9L189,66.9l-57,41.4l21.7,66.9l0,0C187.7,150.6,201.9,106.8,189,66.9	L189,66.9z M39.9,175.2L39.9,175.2l56.9,41.4l56.9-41.4l-56.9-41.4L39.9,175.2z" fill="#FFF"></path>
    </svg>
    Sign In with Auth0
  </amplify-sign-in-button>
</amplify-auth0-button>
`;
